src/software/new_geom/util/closest_point.h

```diff
@@ -12,3 +13,17 @@
  */
 Point closestPointOnLine(const Point &p, const Line &l);
 Point closestPointOnLine(const Line &l, const Point &p);
+
+/**
+ * Finds the Point on line segment closest to point.
+ *
+ * @param centre the point.
+ *
+ * @param segA one end of the line segment.
+ *
+ * @param segB the other end of the line segment.
+ *
+ * @return the Point on line segment closest to centre point.
+ */
+Point closestPointOnSeg(const Point &centre, const Point &segA, const Point &segB);
```

Comment:
```suggestion
Point closestPointOnSeg(const Segment &segment, const Point &p);
```

Reply:
There's already a `Point closestPointOnSeg(const Point &p, const Segment &segment)` on the line below.

Reply:
The interface for closestPointOnSeg should allow (Point,Segment) and (Segment, Point), but not (Point,Point,Point)

Reply:
So your suggestion is to entirely remove the (Point, Point, Point) overload, right?

Reply:
Removed (Point, Point, Point) overload and added (Segment, Point).